reana_workflow_controller/api.py

```diff
@@ -0,0 +1,202 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2017 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 2 of the License, or (at your option) any later
+# version.
+#
+# REANA is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# REANA; if not, write to the Free Software Foundation, Inc., 59 Temple Place,
+# Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+
+"""REANA Workflow Controller Rest API."""
```

Comment:
It may be good to ponder renaming of `api.py` to `restapi.py`, so that we distinguish more clearly here and in other REANA packages between:

- API = Pythonic API, CLI API
- RESTAPI = REST API

While some REANA components won't have much of a Pythonic API, and they will be used exclusively over REST API, it may be good to have some naming consistency across all packages for ease of development across packages?

TBD @diegodelemos @hjhsalo